src/commissaire_http/bus/__init__.py

```diff
@@ -0,0 +1,196 @@
+# Copyright (C) 2016  Red Hat, Inc
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Bus related classes and functions.
+"""
+
+import logging
+import uuid
+
+from kombu import Connection, Exchange, Producer, Queue
+
+
+class Bus:
+    """
+    Connection to a bus.
+    """
+
+    def __init__(self, exchange_name, connection_url, qkwargs):
+        """
+        Initializes a new Bus instance.
+
+        :param exchange_name: Name of the topic exchange.
+        :type exchange_name: str
+        :param connection_url: Kombu connection url.
+        :type connection_url: str
+        :param qkwargs: One or more dicts keyword arguments for queue creation
+        :type qkwargs: list
+        """
+        self.logger = logging.getLogger('Bus')
+        self.logger.debug('Initializing bus connection')
+        self.connection = None
+        self._channel = None
+        self._exchange = None
+        self.exchange_name = exchange_name
+        self.connection_url = connection_url
+        self.qkwargs = qkwargs
+
+    @property
+    def init_kwargs(self):
+        """
+        Returns the initializing kwargs for this instance.
+
+        :returns: the initializing kwargs for this instance.
+        :rtype: dict
+        """
+        return {
+            'exchange_name': self.exchange_name,
+            'connection_url': self.connection_url,
+            'qkwargs': self.qkwargs,
+        }
+
+    def connect(self):
+        """
+        'Connects' to the bus.
+
+        :returns: The same instance.
+        :rtype: commissaire_http.bus.Bus
+        """
```

Comment:
> Would it be useful to make this method idempotent (based on self.connection == None)?

Can't hurt. It's a small change so I'll do it now.

> Should we catch exceptions and reset internal members back to their initial state so we don't leave the bus instance half-connected?

Good idea. Let's tackle that in follow up work.